internal/job/checkout.go

```diff
@@ -549,9 +549,18 @@ func (e *Executor) defaultCheckoutPhase(ctx context.Context) error {
 
 	// Git clean prior to checkout, we do this even if submodules have been
 	// disabled to ensure previous submodules are cleaned up
+	gitSubmodules := false
 	if hasGitSubmodules(e.shell) {
-		if err := gitCleanSubmodules(ctx, e.shell, e.GitCleanFlags); err != nil {
-			return fmt.Errorf("cleaning git submodules: %w", err)
+		// Only clean if submodules have not been explicitly deactivated
```

Comment:
```suggestion
		// Skip cleaning up if submodules have been deactivated
```

The double negative always turns my brain inside out